Backend

artabr opened the pull request:
Two Lambda deployed at:
- https://5dxw106upf.execute-api.eu-central-1.amazonaws.com/dev/products
- https://5dxw106upf.execute-api.eu-central-1.amazonaws.com/dev/products/1

algbeta commented on handler/getProductsById.js:
```diff
@@ -0,0 +1,19 @@
+const serverless = require("serverless-http");
+const express = require("express");
+const app = express();
```
you shouldn't be using express.
 You just describe the handler which should be triggered and the rest is done by aws.

algbeta commented on handler/getProductsList.js:
```diff
@@ -0,0 +1,18 @@
+
+app.use((req, res, next) => {
+    return res.status(404).json({
+        error: "Not Found",
```
it should be `product not found`

algbeta commented on README.md:
```diff
@@ -1,106 +1,13 @@
-After installation, you can start local emulation with:
+Two Lambda deployed at:
+- https://5dxw106upf.execute-api.eu-central-1.amazonaws.com/dev/products
+- https://5dxw106upf.execute-api.eu-central-1.amazonaws.com/dev/products/1
```
would be nice to additionally describe what points were done in README

artabr commented:
@algbeta Please take a look again. All mandatory tasks were done. I removed the Express framework. Additionally, I implemented:
- Async/await
- SWAGGER
- Lambda handlers separated by files
- Main error scenarios are handled by API ("Product not found" error).

The expected score is 8.

algbeta commented on handler/getProductsById.js:
```diff
@@ -0,0 +1,30 @@
+        const product = await fetchProductsById(productId);
+        if (!product) return {
+            statusCode: 404,
+            headers: {
```
just a note: cors could be enabled in serverless.yaml

vkhlystun commented:
Looks good!
